packages/phone-number-privacy/common/src/utils/authentication.ts

```diff
@@ -66,16 +68,26 @@ export function verifyDEKSignature(
   registeredEncryptionKey: string,
   logger?: Logger
 ) {
+  logger = logger ?? rootLogger()
   try {
+    const msgDigest = crypto.createHash('sha256').update(JSON.stringify(message)).digest('base64')
```

Comment:
If I read the `elliptic` docs correctly, the message digest needs to be `hex` encoded. In fact, if I read the code correctly, this will lead to incorrect behavior where it interprets the `base64` as `hex` with strange results. Especially when dealing with a library with such a terrible interface, we need tests that can catch this kind of error.